src/app/tests/TestDataModelSerialization.cpp

```diff
@@ -0,0 +1,872 @@
+        NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+
+        // Ensure that the decoder did not over-write the value set in the generated object
+        NL_TEST_ASSERT(apSuite, t.a == 150);
+
+        NL_TEST_ASSERT(apSuite, t.b == true);
+        NL_TEST_ASSERT(apSuite, t.c == clusters::TestCluster::SimpleEnum::VALUEA);
+
+        for (uint32_t i = 0; i < t.d.size(); i++)
+        {
+            NL_TEST_ASSERT(apSuite, t.d.data()[i] == i);
+        }
+
+        NL_TEST_ASSERT(apSuite, StringMatches(t.e, "chip"));
+    }
+}
+
+void TestDataModelSerialization::TestDataModelSerialization_ExtraField(nlTestSuite * apSuite, void * apContext)
+{
+    CHIP_ERROR err;
+    auto * _this = static_cast<TestDataModelSerialization *>(apContext);
+
+    _this->mpSuite = apSuite;
+    _this->SetupBuf();
+
+    //
+    // Encode
+    //
+    {
+        clusters::TestCluster::SimpleStruct::Type t;
+        uint8_t buf[4]  = { 0, 1, 2, 3 };
+        char strbuf[10] = "chip";
+
+        t.a = 20;
+        t.b = true;
+        t.c = clusters::TestCluster::SimpleEnum::VALUEA;
+        t.d = buf;
+
+        t.e = chip::Span<char>{ strbuf, strlen(strbuf) };
+
+        // Encode every field + an extra field.
+        {
+            err = EncodeStruct(_this->mWriter, TLV::AnonymousTag,
+                               MakeTagValuePair(TLV::ContextTag(clusters::TestCluster::SimpleStruct::kAFieldId), t.a),
+                               MakeTagValuePair(TLV::ContextTag(clusters::TestCluster::SimpleStruct::kBFieldId), t.b),
+                               MakeTagValuePair(TLV::ContextTag(clusters::TestCluster::SimpleStruct::kCFieldId), t.c),
+                               MakeTagValuePair(TLV::ContextTag(clusters::TestCluster::SimpleStruct::kDFieldId), t.d),
+                               MakeTagValuePair(TLV::ContextTag(clusters::TestCluster::SimpleStruct::kEFieldId), t.e),
+                               MakeTagValuePair(TLV::ContextTag(clusters::TestCluster::SimpleStruct::kEFieldId + 1), t.a));
+            NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+        }
+
+        err = _this->mWriter.Finalize();
+        NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+
+        _this->DumpBuf();
+    }
+
+    //
+    // Decode
+    //
+    {
+        clusters::TestCluster::SimpleStruct::Type t;
+
+        _this->SetupReader();
+
+        // Ensure successful decode despite the extra field.
+        err = DataModel::Decode(_this->mReader, t);
+        NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+
+        NL_TEST_ASSERT(apSuite, t.a == 20);
+        NL_TEST_ASSERT(apSuite, t.b == true);
+        NL_TEST_ASSERT(apSuite, t.c == clusters::TestCluster::SimpleEnum::VALUEA);
+
+        for (uint32_t i = 0; i < t.d.size(); i++)
+        {
+            NL_TEST_ASSERT(apSuite, t.d.data()[i] == i);
+        }
+
+        NL_TEST_ASSERT(apSuite, StringMatches(t.e, "chip"));
+    }
+}
+
+struct ByteSpann : public Span<uint8_t>
```

Comment:
`CustomByteSpan` would be clearer that this is not an accidental typo.  Same for the Uint version.

That said, these things look unused.  Just remove them?

Reply:
Cruft as part of my testing/prototyping. Forgot to delete.